src/components/Pagination/Pagination.jsx

```diff
@@ -0,0 +1,208 @@
+  const [currentPage, setCurrentPage] = useState(1);
+  const [pages, setPages] = useState([]);
+
+  useEffect(() => {
+    const { totalRecords, pageLimit, pageNeighbors } = props;
+
+    if (pageLimit) {
+      setPageLimit(pageLimit);
+    }
+    if (pageNeighbors) {
+      setPageNeighbors(Math.max(0, Math.min(pageNeighbors, 2)));
+    }
+
+    const totalPages = Math.ceil(totalRecords / pageLimit);
+    setTotalRecords(totalRecords);
+    setTotalPages(totalPages);
+  }, []);
+
+  useEffect(() => {
+    setPages(fetchPageNumbers());
+  }, [totalPages, currentPage]);
+
+  const range = (from, to, step = 1) => {
+    let i = from;
+    const range = [];
+
+    while (i <= to) {
+      range.push(i);
+      i += step;
+    }
+
+    return range;
+  };
+
+  /**
+   * Let's say we have 10 pages and we set pageNeighbours to 2
+   * Given that the current page is 6
+   * The pagination control will look like the following:
+   *
+   * (1) < {4 5} [6] {7 8} > (10)
+   *
+   * (x) => terminal pages: first and last page(always visible)
+   * [x] => represents current page
+   * {...x} => represents page neighbours
+   */
+  const fetchPageNumbers = () => {
+    /**
+     * totalNumbers: the total page numbers to show on the control
+     * totalBlocks: totalNumbers + 2 to cover for the left(<) and right(>) controls
+     */
+    const totalNumbers = pageNeighbors * 2 + 3;
+    const totalBlocks = totalNumbers + 2;
+
+    if (totalPages > totalBlocks) {
+      const startPage = Math.max(2, currentPage - pageNeighbors);
+      const endPage = Math.min(totalPages - 1, currentPage + pageNeighbors);
+
+      let pages = range(startPage, endPage);
+
+      /**
+       * hasLeftSpill: has hidden pages to the left
+       * hasRightSpill: has hidden pages to the right
+       * spillOffset: number of hidden pages either to the left or to the right
+       */
+      const hasLeftSpill = startPage > 2;
+      const hasRightSpill = totalPages - endPage > 1;
+      const spillOffset = totalNumbers - (pages.length + 1);
+
+      switch (true) {
+        // handle: (1) < {5 6} [7] {8 9} (10)
+        case hasLeftSpill && !hasRightSpill: {
+          const extraPages = range(startPage - spillOffset, startPage - 1);
+          pages = [LEFT_PAGE, ...extraPages, ...pages];
+          break;
+        }
+
+        // handle: (1) {2 3} [4] {5 6} > (10)
+        case !hasLeftSpill && hasRightSpill: {
+          const extraPages = range(endPage + 1, endPage + spillOffset);
+          pages = [...pages, ...extraPages, RIGHT_PAGE];
+          break;
+        }
+
+        // handle: (1) < {4 5} [6] {7 8} > (10)
+        case hasLeftSpill && hasRightSpill:
+        default: {
+          pages = [LEFT_PAGE, ...pages, RIGHT_PAGE];
+          break;
+        }
+      }
+
+      return [1, ...pages, totalPages];
+    }
+
+    return range(1, totalPages);
+  };
+
+  const gotoPage = page => {
+    const { onPageChanged = f => f } = props;
```

Comment:
This should be in an if-statement below instead, no need to make a new function.

```
if (onPageChanged) onPageChanged(paginationData)
```